### PR TITLE
Update to OrientDB 2.1-rc4

### PR DESCRIPTION
**Pull request by mcculls**
Also updated NuGet IT expectation to account for slight difference in sorting of text fields.
Text fields are sorted by dictionary order in 2.1.x using Java's Collator (rather than naive
sorting using String.compareTo) which means that 'bar' now appears before 'Foo'.

http://bamboo.s/browse/NX3-OSSF456-1 :white_check_mark: 

Haven't been able to recreate the issue we saw with 2.1-rc3 so will merge this later today.

### CODE REVIEW

**Comment by cstamas** [comment]
+1

File: testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetProxyIT.java [hunk 1]
```diff
@@ -132,7 +132,8 @@ public void sortOrderIsRespected() throws Exception {
 
     List<String> receivedHashes = extractPackageHashes(entries);
 
-    assertThat(receivedHashes, is(sorted(receivedHashes)));
+    Collator dictionaryOrder = Collator.getInstance(Locale.ENGLISH);
+    assertThat(receivedHashes, is(Ordering.from(dictionaryOrder).sortedCopy(receivedHashes)));
   }
```
Comment (bradbeck) [inline testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetProxyIT.java]: `@SuppressWarnings("unused")` would remove the compiler warning for `hashOrderedEntries` and make it clearer that it is intentional.

**Comment by bradbeck** [comment]
+1 aside from the compiler warning comment.

File: testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetProxyIT.java (hunk 1 above)
Comment (mcculls) [inline testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetProxyIT.java]: sure, that's best addressed in a separate PR though given that `hashOrderedEntries` is not part of this PR and there are plenty of other tests that need cleaning up wrt. unused compiler warnings.

**Comment by mcculls** [comment]
Merged to master as 28c3b3e
